Ira/my-app-video/src/redux/movies/movieSaga.ts

```diff
@@ -0,0 +1,78 @@
+import {takeLatest, call, put, all} from 'redux-saga/effects';
+import {MovieActionTypes} from "./actionTypes";
+import {baseUrl} from "../../constants/baseUrl";
+import {
+  fetchMoviesSuccess,
+  fetchMoviesError,
+  IFetchMovies,
+  IFetchMovieById,
+  fetchMovieByIdSuccess,
+} from "./actions";
+
+function* fetchMoviesSaga(action: IFetchMovies) {
+  const limit = 9;
+  const page = (action.payload?.page || 1) - 1;
+  const offset: number = page * limit + 1;
+  let url = `${baseUrl}/movies?limit=${limit}&offset=${offset}&sortOrder=desc`;
+
+  if (action.payload === undefined) {
+    url += `&sortBy=release_date`;
+  }
+
+  if (action.payload?.search) {
+    url += `&search=${action.payload?.search}`;
+  }
+
+  if (action.payload?.searchBy) {
+    url += `&searchBy=${action.payload?.searchBy}`;
+  }
+
+  if (action.payload?.sortBy) {
+    url += `&sortBy=${action.payload?.sortBy}`;
+  }
+  
+  if (action.payload?.searchBy === 'title') {
+    url = `${baseUrl}/movies?limit=${limit}&search=${action.payload?.search}&searchBy=${action.payload?.searchBy}`;
+  }
```

Comment:
you can filter empty values or use querystring.stringify it removes empty values automaticaly